beetsplug/importreplace.py

```diff
@@ -68,9 +68,11 @@ def _albuminfo_received(self, info: AlbumInfo):
         for track in info.tracks:
             self._trackinfo_received(track)
 
-    def _replace_field(self, text: str, replacements: [(Pattern, str)]) -> str:
-
-        return reduce(self._replace, replacements, text)
+    def _replace_field(self, text: str|list, replacements: [(Pattern, str)]) -> str|list:
+        if isinstance(text, list):
+            return list(map(lambda item: reduce(self._replace, replacements, item), text))
```

Comment:
I think `lambda item: self._replace_field(item, replacements)` would be nicer, since it'd keep the same logic as the `str` case.

Reply:
I think this may be a nice application for `@singledispatchmethod`, something like

```python
    @singledispatchmethod
    @classmethod
    def _replace(cls, value: Any, replacement: Replacement) -> Any:
        raise NotImplementedError

    @_replace.register(str)
    @classmethod
    def _replace_str(cls, value: str, replacement: Replacement) -> str:
        return replacement[0].sub(repl=replacement[1], string=value)

    @_replace.register(list)
    @classmethod
    def _replace_list(cls, value: list[str], replacement: Replacement) -> list[str]:
        return [cls._replace(item, replacement) for item in value]
```

Reply:
It looks like `@singledispatchmethod` was introduced in [3.4](https://docs.python.org/3.4/library/functools.html#functools.singledispatch).

@edgars-supe, would you have an opinion on the minimum version of Python we would want to support?

I appreciate all of these comments and suggestions, I am learning a lot :)